File: 백준/Bronze/11328. Strfry/Strfry.swift
import Foundation

let n = Int(readLine()!)!
for _ in 0..<n {
    let ab = readLine()!.split(separator: " ").map{String($0)}
    
    let a = ab[0].map{String($0)}
    let b = ab[1].map{String($0)}
    
    var a_alp = Array(repeating: 0, count: 26)
    var b_alp = Array(repeating: 0, count: 26)

    if (a.count != b.count) {
        print("Impossible")
        
    } else {
        for c in a {
            let idx = Int(UnicodeScalar(c)!.value)-97
            a_alp[idx] += 1
        }
    
        for c in b {
            let idx = Int(UnicodeScalar(c)!.value)-97
            b_alp[idx] += 1
        }
    
        var answer = "Possible"
        for i in 0..<26 {
            if (a_alp[i] != b_alp[i]) { 
                answer = "Impossible"
                break
            }
        }
        print(answer)
    }
}

File: 백준/Bronze/11328. Strfry/Strfry.java
import java.util.*;
import java.lang.*;
import java.io.*;

class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());

        for(int i=0; i<n; i++) {
            String s = br.readLine();
            StringTokenizer st = new StringTokenizer(s);
            
            char[] a = st.nextToken().toCharArray();
            char[] b = st.nextToken().toCharArray();

            if (a.length != b.length) System.out.println("Impossible");
            else {
                Arrays.sort(a);
                Arrays.sort(b);
                
                String answer = "Possible";
                for(int j=0; j<a.length; j++) {
                    if(a[j] != b[j]) answer = "Impossible";
                }
                System.out.println(answer);
            }
        }
        
    }
}